src/shapleypy/classes/checkers.py

```diff
@@ -0,0 +1,61 @@
+# ruff: noqa: N806
+from __future__ import annotations
+
+from shapleypy.coalition import (
+    EMPTY_COALITION,
+    Coalition,
+    all_one_player_missing_subcoalitions,
+)
+from shapleypy.game import Game
+
+
+def check_monotonicity(game: Game) -> bool:
+    """
+    Check if the game is monotone.
+    """
+    for S in game.all_coalitions:
+        # Check the values of all subcoalitions of size |S| - 1
+        value_of_S = game.get_value(S)
+        for S_minus_one in all_one_player_missing_subcoalitions(S):
+            if value_of_S < game.get_value(S_minus_one):
+                return False
+    return True
+
+
+def check_weakly_superadditivity(game: Game) -> bool:
+    grand_coalition = Coalition.grand_coalition(game.number_of_players)
+    for i in range(game.number_of_players):
+        value_of_i = game.get_value([i])
+        for S in (grand_coalition - i).all_subcoalitions():
+            if game.get_value(S) + value_of_i > game.get_value(S + i):
+                return False
+    return True
+
+
+def check_superadditivity(game: Game) -> bool:
+    for T in game.all_coalitions:
+        value_of_T = game.get_value(T)
+        for S in filter(
+            lambda s: T * s == EMPTY_COALITION, game.all_coalitions
+        ):
+            if value_of_T + game.get_value(S) > game.get_value(T + S):
+                return False
+    return True
+
+
+def check_convexity(game: Game) -> bool:
```

Comment:
There is an alternative definition of convexity, stating that for every i,j \in N and every S \subseteq N - {i,j}, the convexity condition has to be satisfied, v(S + {i,j}) - v(S + {i}) >= v(S + j) - v(S).

This should result in a faster checker than we have implemented here. I do not know how fast it is for larger number of players, but if it was a significant speed-up, we should go for it.

Reply:
Added in [adac682](https://github.com/muzikr/Shapleypy/pull/21/commits/adac6824ba58afac995425cb65016f28dcfbb7a7)